### PR TITLE
Fix nonexisting groups errors and replace stage labels with groups

### PR DESCRIPTION
**Pull request by lgalis**
Replace 'Stage' with 'Group' labels in the Add/edit workflow menu and modals.

https://projects.engineering.redhat.com/browse/SSP-995

Also updated the error handling for the groups that was causing the workflows list to not display.

### CODE REVIEW

**Comment by gmcculloug** [comment]
@rvsia Can you help review this PR?

File: src/helpers/group/group-helper.js [hunk 1]
```diff
@@ -10,8 +10,17 @@ export async function getRbacGroups() {
 export async function fetchGroupNames(groupRefs) {
   if (groupRefs) {
     return Promise.all(groupRefs.map(async id => {
-      const group = await api.getGroup(id);
-      return group.name;
+      try {
+        const group = await api.getGroup(id);
+        return group.name;
+      } catch (error) {
+        if (!(error && error.status === 404)) {
```
Comment (rvsia) [inline src/helpers/group/group-helper.js]: `error.status !== 404` should be enough

File: src/smart-components/workflow/add-stages/add-stages-wizard.js [hunk 2]
```diff
@@ -25,7 +25,7 @@ const AddWorkflow = ({
 
   const steps = [
     { name: 'General Information', component: <StageInformation formData={ formData } handleChange={ handleChange } /> },
-    { name: 'Set Stages', component: <SetStages formData={ formData }
+    { name: 'Set Groups', component: <SetStages formData={ formData }
```
Comment (Hyperkid123) [inline src/smart-components/workflow/add-stages/add-stages-wizard.js]: We should use sentence case `Set groups`. Can you fix it here or do you want to do these in separate PR?

Comment (lgalis) [reply src/smart-components/workflow/add-stages/add-stages-wizard.js]: I'll fix it here.

**Comment by lgalis** [comment]
@Hyperkid123 - feedback addressed
